docs/changelog.jl

```diff
@@ -0,0 +1,134 @@
+    content = read(changelog_inputfile, String)
+    content = first(split(content, CHANGELOG_LINK_SEPARATOR))
+
+    # Rule: [abc#XXXX] -> https://github.com/abc/issues/XXXX
+    # Description: Replace issue/PR numbers with a link to the default repo
+    # Example: [JuliaLang/julia#123] -> https://github.com/JuliaLang/julia/issues/123
+    # There is no need to distinguish between PRs and Issues because GitHub redirects.
+    for m in eachmatch(r"\[(?<repo>[a-zA-Z0-9/]+?)\#(?<id>[0-9]+)\]", content)
+        linkmap[m.match] = "https://github.com/$(m["repo"])/issues/$(m["id"])"
+    end
+
+    # Rule: [#XXXX] -> https://github.com/url/issue/XXXX
+    # Description: Replace issue/PR numbers with a link to the default repo
+    # Example: [#123] -> https://github.com/JuliaDocs/Documenter.jl/issues/123
+    # There is no need to distinguish between PRs and Issues because GitHub redirects.
+    for m in eachmatch(r"\[\#(?<id>[0-9]+)\]", content)
+        linkmap[m.match] = "https://github.com/$(repo)/issues/$(m["id"])"
+    end
+
+    # Rule: [@XXXX] -> https://github.com/XXXX
+    # Description: Replace users with a link to their GitHub
+    # Example: [@odow] -> https://github.com/odow
+    for m in eachmatch(r"\[@(?<id>.+?)\]", content)
+        linkmap[m.match] = "https://github.com/$(m["id"])"
+    end
+
+    # Rule: ## Version [vX.Y.Z] -> url/releases/tag/vX.Y.Z
+    # Description: Replace version headers with a link to the GitHub release
+    # Example: ## Version [v0.27.0] -> https://github.com/JuliaDocs/Documenter.jl/releases/tag/v0.27.0
+    for m in eachmatch(r"^\#\# Version (?<token>\[(?<tag>v[0-9]+.[0-9]+.[0-9]+)\])"m, content)
+        linkmap[m["token"]] = "https://github.com/$(repo)/releases/tag/$(m["tag"])"
+    end
+
+    return linkmap
+end
+
+# Rewrite CHANGELOG.md to a Documenter friendly markdown file
+function rewrite_changelog(;
+    changelog_inputfile::String,
+    changelog_outputfile::String,
+    # current_module::String,
+    repo::String,
+    branch::String = "master",
+)
+    # Get the map of token to full URL
+    linkmap = collect_links(; changelog_inputfile, repo)
+
+    # Read the source file and split the content to ignore the list of links
+    content = read(changelog_inputfile, String)
+    content = first(split(content, CHANGELOG_LINK_SEPARATOR))
+
+    # Replace all link tokens with full URLs
+    for (k, v) in linkmap
+        content = replace(content, k => "$k($v)")
+    end
+
+    # Header to set EditURL
+    header = """
+    ```@meta
+    EditURL = "https://github.com/$repo/blob/$branch/CHANGELOG.md"
+    ```
+    """
+
+    # Write it all out
+    open(changelog_outputfile, "w") do io
+        write(io, header)
+        write(io, content)
+    end
+
+    return
+end
+
+# Collect and add all links to the CHANGELOG.md footer.
+# julia -L docs/changelog.jl -e 'write_link_footer(; changelog_inputfile = "CHANGELOG.md", repo = "JuliaDocs/Documenter.jl")'
```

Comment:
Any thoughts on having like an `isinteractive() || main(ARGS)` type pattern, so that you wouldn't have to copy this whole thing into the terminal? Although that would clash with the `include` in `make.jl` though (in case that file is loaded in an interactive session)..

Reply:
`make changelog` seems easier.